### PR TITLE
Feature/extracting catalog modules

### PR DESCRIPTION
**Pull request by sergy-gzh**
### Please check if the PR fulfills these requirements ###

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
All modifications to magento standard /products API removed.
Introduced new category/:categoryId/products API instead
Fixed some issue to enable swagger.
Removed some customizations
**What is the current behavior? (You can also link to an open issue here)**
Old API was a massively rewritten magento standard `products` API.
Issues with old API:
- was not taking use of magento indexes for categories/products/filters
- Had SQL queries directly attached to magento collection, which makes future extension problematic
**What is the new behavior (if this is a feature change)?**
New API provide similar output as Modified `products` did.
Response contains product list data with filters
**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**
Yes,
**Other information:**

### CODE REVIEW

File: app/code/Deity/Catalog/composer.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    "name": "deity/falcon-magento-catalog",
+    "description": "Deity Magento Catalog API implementation",
+    "type": "magento2-module",
+    "version": "1.0.0",
+    "require": {
+        "magento/framework": "*",
+        "magento/module-catalog" : "*",
```
Comment (schmengler) [inline app/code/Deity/Catalog/composer.json]: any plans to test it against different versions to determine compatibility?

File: app/code/Deity/Theme/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+### Deity Theme module
```
Comment (schmengler) [inline app/code/Deity/Theme/README.md]: The purpose of this module would be interesting to document here. Do you need to configure a Magento Theme for Deity?

File: app/code/Deity/Catalog/composer.json (hunk 1 above)
Comment (sergy-gzh) [reply app/code/Deity/Catalog/composer.json]: so far we only support magento >= 2.2
I don't see any value of testing particular module versions, cause they increase along with magento product version.
so for Magento 2.1 -> catalog 101.*
magento 2.2 -> catalog 102.*
magento 2.3 -> catalog 103.*

File: app/code/Deity/Theme/README.md (hunk 2 above)
Comment (sergy-gzh) [reply app/code/Deity/Theme/README.md]: Will update the description.
Magento is absolutely headless in this setup.
However some resources are still generated by it.
Such as resized product images in this case.
By default its preconfigured for our default theme.

File: app/code/Deity/Catalog/composer.json (hunk 1 above)
Comment (schmengler) [reply app/code/Deity/Catalog/composer.json]: What I meant is: add constraints here that match the Magento versions you support, e.g. `"magento/module-catalog": "^101.0.0|^102.0.0"` if 2.1.x and 2.2.x are supported (or whatever the minimum catalog module version in each minimum supported Magento version is)

Integration / API tests can still run against the latest version of each 2.1 / 2.2 / 2.3 branch

Comment (schmengler) [reply app/code/Deity/Catalog/composer.json]: But knowing what is supported requires to test against specific versions, that's why I asked

Comment (sergy-gzh) [reply app/code/Deity/Catalog/composer.json]: @schmengler we have those constraints on package level.
I don't see a need to specify strict dependencies per-module cause we have it globally per package.
Magento module versions correlate with platform version.
We also run test suite for all modules.

File: app/code/Deity/Catalog/Model/ProductConvert.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    {
+        Profiler::start('__PRODUCT_LISTING_CONVERT__', ['group' => 'Deity']);
+
+        $this->currentProductObject = $product;
+
+        $deityProduct = $this->productFactory->create();
+        $deityProduct->setName($product->getName());
+        $deityProduct->setSku($product->getSku());
+        $deityProduct->setUrlPath($this->getProductUrlPath($product));
+        $deityProduct->setIsSalable((int)$product->getIsSalable());
```
[the remark this answers is not in the document]
Comment (sergy-gzh) [reply app/code/Deity/Catalog/Model/ProductConvert.php]: @schmengler I have introduced example plugin on how to add extra attributes for product API:
see: app/code/Deity/Catalog/Plugin/ProductConvertPluginExample.php

File: app/code/Deity/Catalog/Model/Data/Filter.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+use Deity\CatalogApi\Api\Data\FilterInterface;
+use Magento\Framework\Api\AbstractSimpleObject;
+use Magento\Framework\Phrase;
+
+class Filter extends AbstractSimpleObject implements FilterInterface
+{
+    /**
+     * @return string
+     */
+    public function getLabel()
```
[the remark this answers is not in the document]
Comment (sergy-gzh) [reply app/code/Deity/Catalog/Model/Data/Filter.php]: true, fixed

File: app/code/Deity/Catalog/Model/Product/ImageProvider.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * @param Product $product
+     * @param string $imageType
+     * @return string
+     */
+    public function getProductImageTypeUrl(Product $product, string $imageType): string
+    {
+        $this->appEmulation->startEnvironmentEmulation($product->getStoreId(), Area::AREA_FRONTEND, true);
+        // TODO: Implement getProductImageTypeUrl() method.
```
[the remark this answers is not in the document]
Comment (sergy-gzh) [reply app/code/Deity/Catalog/Model/Product/ImageProvider.php]: fixed

File: app/code/Deity/Catalog/Model/Product/ImageProvider.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    /**
+     * @param Product $product
+     * @param string $imageType
+     * @return string
+     */
+    public function getProductImageTypeUrl(Product $product, string $imageType): string
+    {
+        $this->appEmulation->startEnvironmentEmulation($product->getStoreId(), Area::AREA_FRONTEND, true);
```
[the remark this answers is not in the document]
Comment (sergy-gzh) [reply app/code/Deity/Catalog/Model/Product/ImageProvider.php]: good point.

**Pull request merged**